lib/THC/THCApply.cuh

```diff
@@ -39,6 +39,71 @@ kernelPointwiseApply1(TensorInfo<Ta, IndexType> a,
 }
 
 template <typename Op,
+          typename Ta,
```

Comment:
linspace and logspace are 1d and should only really operate on contiguous 1-d tensors, so there's no need to add a version that passes in an index to the operator. This is just duplicating a bunch of code. I'd remove these and just go the simple route.

Instead, you can just add the base pointer to the operator instance itself and your operator can calculate the offset.

A multi-dimensional linear index is not that useful in general to operators, since often you would need the dimension information to make sense of it if you have more than one dimension.